### PR TITLE
Todo functions branch

### PR DESCRIPTION
**Pull request by App-Developer8866**
Following changes:

- Add function
- Remove function
-

### CODE REVIEW

File: modules/addToList.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import updateBookData from './updateBookData.js';
+import displayLists from './displayLists.js';
+
+export default function addToList(value, completed, index, toDoList) {
+  toDoList.push({ value, completed, index });
```
Comment (Shedrack-Sunday) [inline modules/addToList.js]: - [ ] ES6 improves JavaScript by introducing new features such as arrow functions, template literals, and classes, which enhance code readability and maintainability. It also provides modules for better organization and code reusability in larger projects, promoting collaboration among developers.
- Kindly refactor your code to use `ES6` 👍

File: modules/clearList.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import updateBookData from './updateBookData.js';
+import displayLists from './displayLists.js';
+
+export default function clearList(taskList) {
```
Comment (Shedrack-Sunday) [inline modules/clearList.js]: - [ ] Kindly check my comment on the use of `es6` on the addToList file which is very detailed 👍

File: modules/displayLists.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import removeListitem from './removeListitem.js';
+import editList from './editList.js';
+import markComplete from './markComplete.js';
+import markUnComplete from './markUnComplete.js';
+
+export default function displayLists(taskList) {
```
Comment (Shedrack-Sunday) [inline modules/displayLists.js]: - [ ]  Kindly check my comment on the use of es6 on the addToList file which is very detailed 👍

File: modules/editList.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import updateBookData from './updateBookData.js';
+
+export default function editList(index, value, toDoList) {
```
Comment (Shedrack-Sunday) [inline modules/editList.js]: - [ ]  Kindly check my comment on the use of es6 on the addToList file which is very detailed 👍

File: modules/markComplete.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import updateStatus from './updateStatus.js';
+
+export default function markComplete(event, toDoList) {
+  const checkbox = event.target;
```
Comment (Shedrack-Sunday) [inline modules/markComplete.js]: - [ ]  Kindly check my comment on the use of es6 on the addToList file which is very detailed 👍

File: modules/markUnComplete.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import updateStatus from './updateStatus.js';
+
+export default function markUnComplete(event, toDoList) {
+  const checkbox = event.target;
```
Comment (Shedrack-Sunday) [inline modules/markUnComplete.js]: -  [ ]  Kindly check my comment on the use of es6 on the addToList file which is very detailed 👍

File: modules/refreshList.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import displayLists from './displayLists.js';
+
+export default function refreshList(toDoList) {
```
Comment (Shedrack-Sunday) [inline modules/refreshList.js]: - [ ]  Kindly check my comment on the use of es6 on the addToList file which is very detailed 👍

File: modules/removeListitem.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import updateBookData from './updateBookData.js';
+
+export default function removeListitem(index, toDoList) {
```
Comment (Shedrack-Sunday) [inline modules/removeListitem.js]: - [ ]  Kindly check my comment on the use of es6 on the addToList file which is very detailed 👍

File: modules/updateBookData.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+export default function updateBookData(collectionData) {
+  localStorage.setItem('tasksList', JSON.stringify(collectionData));
+  const tasksList = JSON.parse(localStorage.getItem('tasksList'));
```
Comment (Shedrack-Sunday) [inline modules/updateBookData.js]: - [ ]  Kindly check my comment on the use of es6 on the addToList file which is very detailed 👍

File: modules/updateLists.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import removeListitem from './removeListitem.js';
+import displayLists from './displayLists.js';
+
+export default function updateLists(toDoList) {
+  displayLists(toDoList);
```
Comment (Shedrack-Sunday) [inline modules/updateLists.js]: - [ ]  Kindly check my comment on the use of es6 on the addToList file which is very detailed 👍

File: modules/updateLists.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+  const removeTaskIcons = document.querySelectorAll('.list-icon');
+  function handleRemoveTaskClick(event) {
+    const listItem = event.target.closest('li');
```
Comment (Shedrack-Sunday) [inline modules/updateLists.js]: - [ ]  Kindly check my comment on the use of es6 on the addToList file which is very detailed 👍

File: modules/updateStatus.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import updateBookData from './updateBookData.js';
+
+export default function updateStatus(index, status, toDoList) {
```
Comment (Shedrack-Sunday) [inline modules/updateStatus.js]: - [ ]  Kindly check my comment on the use of es6 on the addToList file which is very detailed 👍

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello  @App-Developer8866 👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️  
- Kudus on the use of modules✔️ 
  
# Required Changes  🌵 

- [ ] You made some impressive implementations which I must say is a job well done. However, there are a few  I suggest   you make In line with the requirement

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: modules/markUnComplete.js (hunk 6 above)
Comment (DuaneDave) [inline modules/markUnComplete.js]: As previously mentioned and as part of the project requirements, consider refactoring this function to use es6 arrow function syntax for readability and to satisfy the milestone requirements

File: modules/refreshList.js (hunk 7 above)
Comment (DuaneDave) [inline modules/refreshList.js]: As previously mentioned, and as part of the project requirements, consider refactoring this function to use es6 arrow function syntax for readability and to satisfy the milestone requirements.

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @App-Developer8866 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ The way your code is modularized is quite intuitive and great

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @App-Developer8866 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

CookedDelectableDolphin-size_restricted


## Highlights
Used modules :+1: 
Used arrow functions :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**